src/ui.js

```diff
@@ -4350,6 +4362,25 @@ class Panel {
                         return output;
                     });
                 })
+        ).add(
+          // The embed button.
+          new DOM.Element("button").add("Get embed code")
+              .listen("click", () => {
+                  display_export_pane("base64", (output) => {
+                      const URL_prefix = window.location.href
+                          .replace(/(\/index\.html)?\?.*$/, "")
+                      return {
+                          data: `<!-- ${URL_prefix} -->
+<div class="quiver-embed">
+    <iframe src="${output.data}&macro_url=${
+        encodeURIComponent(ui.macro_url)
+    }&e=" width="400" height="400"></iframe>
```

Comment:
It would be nice to vary the width and height based on the content of the diagram, so that there are always borders of a fixed size around the diagram. You can get the grid column and row sizes with `ui.cell_width` and `ui.cell_height` and use similar logic to below to iterate through each of the cells to find the total size of the diagram.
https://github.com/varkor/quiver/blob/b8c1ba4333fc584b7c7ec062ab9fabb4d88f9a58/src/quiver.js#L253-L261

Reply:
I thought, if anything, it would be nice to set the scale dynamically, rather than the width and height. When I go to most websites offering a widget, I know that the widget I'm going to get back is always going to be the same size, and if I want to adjust it, I know where to look.

Reply:
If you think that's a good idea, I'm happy to take a look at it, otherwise I'll have a crack at setting the width and height as you've said.

Reply:
That's a good point. I could potentially see either choice being useful. In that case, it would be ideal to let the user decide, with an option (like "Centre diagram" when exporting to LaTeX):
![image](https://user-images.githubusercontent.com/3943692/106765022-48c0f280-6630-11eb-807e-4987e460072f.png)
If there were also width and height inputs in that options bar (the values for which were saved in `localStorage` alongside the "Centre diagram option"), then the user would have full control of the output, and we could also set the zoom level to fit the diagram regardless of its size.
https://github.com/varkor/quiver/blob/b8c1ba4333fc584b7c7ec062ab9fabb4d88f9a58/src/ui.js#L4284
This seems more useful than picking an arbitrary size that the user might have to manually change in the output.

If the "Fixed size" checkbox was not checked, then it could output at the size of the diagram itself (the same logic is necessary to calculate this size and also the zoom level).

Reply:
This sounds perfect, I hadn't realise you'd already got the framework set up for adding options to exports. I'll get on this right away!

Reply:
I think I'm pretty much done. Unfortunately, I don't think we'll be able to support the `fixed size` check box without giving `QuiverExport` the ability to look at the `ui` which doesn't seem pleasant. User input width and height is implemented though and I also managed to get the auto-scaling working properly.

Reply:
(As in, we won't be able to support having `fixed size` unchecked as otherwise the export code will have to run `ui.diagram_size()`)

Reply:
I'm going to play around with the implementation and maybe make a few tweaks. I'll also think about this problem with no fixed size.